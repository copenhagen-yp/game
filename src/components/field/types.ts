import { ReactNode } from 'react';

export type Props = {
    children: ReactNode,
    label?: string,
    isError?: boolean,
    className?: string
};