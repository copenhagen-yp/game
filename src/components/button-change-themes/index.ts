export * from './button-change-themes';
