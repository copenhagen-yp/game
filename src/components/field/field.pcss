.field {
    position: relative;
    margin: 0 30px;
}

.field__label {
    position: absolute;
    font-size: 9px;
    top: 0;
    left: 0;
    margin-bottom: 10px;
    color: #999999;
}

.error {
    position: absolute;
    bottom: 0;
    left: 0;
    color: #FF2F2F;
    font-size: 10px;
}