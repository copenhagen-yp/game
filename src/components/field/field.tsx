import React, { FC } from 'react';
import classNames from 'classnames';

import { Props } from './types';
import styles from './field.pcss';

export const Field: FC<Props> = (props) => {
  const { children, label, isError, className } = props;
    
  return (
    <div className={classNames(styles.field, className)}>
      <label className={styles.field__label}>{label}</label>
      {children}
      {isError ? <span className={styles.error}>Поле обязательно для заполнения</span> : null}
    </div>
  )
};
