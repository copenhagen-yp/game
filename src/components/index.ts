import { Button } from './button';
import { Toast } from './toast';
import { Input } from './input';

export {
  Button,
  Toast,
  Input
};