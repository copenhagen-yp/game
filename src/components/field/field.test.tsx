import React from 'react';
import renderer from 'react-test-renderer';
import { Field } from './field';
import { Input } from '../../components';

test('Field renders correctly', () => {
  const tree = renderer
    .create(<Field>
      <Input
        type='password'
        name='password'
      />
    </Field>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
