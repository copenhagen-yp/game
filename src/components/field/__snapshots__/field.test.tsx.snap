// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field renders correctly 1`] = `
<div
  className="field"
>
  <label
    className="field__label"
  />
  <input
    className="input"
    name="password"
    type="password"
  />
</div>
`;
